cmd/terramate/cli/cli.go

```diff
@@ -935,40 +934,28 @@ func (c *cli) runOnStacks() {
 
 	logger.Info().Msg("Running on selected stacks")
 
-	failed := false
-
-	for _, stack := range orderedStacks {
-		cmd := exec.Command(c.parsedArgs.Run.Command[0], c.parsedArgs.Run.Command[1:]...)
-		cmd.Dir = stack.HostPath()
-		cmd.Env = os.Environ()
-		cmd.Stdin = c.stdin
-		cmd.Stdout = c.stdout
-		cmd.Stderr = c.stderr
-
-		logger := log.With().
-			Str("cmd", strings.Join(c.parsedArgs.Run.Command, " ")).
-			Stringer("stack", stack).
-			Logger()
+	err = run.Exec(
+		orderedStacks,
+		c.parsedArgs.Run.Command,
+		c.stdin,
+		c.stdout,
+		c.stderr,
+		c.parsedArgs.Run.ContinueOnError,
+	)
 
-		logger.Info().Msg("Running")
+	if err != nil {
 
-		err = cmd.Run()
-		if err != nil {
-			failed = true
+		logger.Warn().Msg("one or more commands failed")
 
-			if c.parsedArgs.Run.ContinueOnError {
-				logger.Warn().
-					Err(err).
-					Msg("failed to execute command")
-			} else {
-				logger.Fatal().
-					Err(err).
-					Msg("failed to execute command")
+		var errs *errors.List
+		if errors.As(err, &errs) {
+			for _, err := range errs.Errors() {
+				logger.Warn().Err(err).Send()
```

Comment:
here there is a slight change in behavior, before when there was no continue on error the message would be fatal. Now the message is always a warning and then we exit with an error status code. Since exiting with the err status code is the most important part it seems this change is OK, given that it simplifies the handling of the errors (doesn't matter if it is one or N errors).